common/cpp/src/google_smart_card_common/formatting.cc

```diff
@@ -48,4 +49,15 @@ std::string FormatPrintfTemplate(const char* format, va_list var_args) {
   }
 }
 
+void FormatPrintfTemplateAndSet(std::string* output_string, const char* format,
+                                ...) {
+  va_list var_args;
+  va_start(var_args, format);
+  std::string formatted = FormatPrintfTemplate(format, var_args);
+  va_end(var_args);
+  // Note: It's incorrect to move this check to the beginning of the function,
+  // since we need to read the variable arguments off the stack in any case.
+  if (output_string) *output_string = std::move(formatted);
```

Comment:
nit: maybe return early if output_string is null already to avoid doing the above logic regardless.

Reply:
Done. I actually thought it's illegal to do so (assuming that the stack would be corrupted if we don't ask `printf()` to read all arguments), but after some googling it seems to be safe to do so.
I've also added a unit test for this, and it passes successfully.